File: app/src/main/java/luyao/androidutils/MainActivity.java
package luyao.androidutils;

import android.Manifest;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.Environment;
import android.support.v4.app.ActivityCompat;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import java.io.File;
import java.util.Arrays;

import luyao.lib.DeviceUtils;
import luyao.lib.encrypt.RC4Utils;
import luyao.lib.storage.FileUtils;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        File file = new File(Environment.getExternalStorageDirectory().getPath());
        Log.e("cache", FileUtils.getFileSize(file));

        String data = "你好";
        String key = "123456";

        byte[] enc = RC4Utils.rc4(data.getBytes(), key.getBytes());
        Log.e("jni enc", Arrays.toString(enc));
        byte[] dec = RC4Utils.rc4Native(enc, key.getBytes());
//        byte[] dec2 = RC4Utils.rc4(dec, key.getBytes());

        Log.e("jni enc", Arrays.toString(dec));
        Log.e("jni", new String(dec));

        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.READ_PHONE_STATE) != PackageManager.PERMISSION_GRANTED) {
            // TODO: Consider calling
            //    ActivityCompat#requestPermissions
            // here to request the missing permissions, and then overriding
            //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
            //                                          int[] grantResults)
            // to handle the case where the user grants the permission. See the documentation
            // for ActivityCompat#requestPermissions for more details.
            return;
        }
        String imei = DeviceUtils.getIMEI(getApplicationContext());
        Log.e("imei", imei);
    }
}

File: app/src/main/java/luyao/androidutils/MainActivity.kt
package luyao.androidutils

import android.os.Bundle
import android.os.Environment
import android.support.v7.app.AppCompatActivity
import android.util.Log
import luyao.lib.encrypt.RC4Utils
import luyao.lib.storage.FileUtils
import java.io.File
import java.util.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val file = File(Environment.getExternalStorageDirectory().path)
        Log.e("cache", FileUtils.getFileSize(file))

        val data = "你好"
        val key = "123456"

        val enc = RC4Utils.rc4(data.toByteArray(), key.toByteArray())
        Log.e("jni enc", Arrays.toString(enc))
        val dec = RC4Utils.rc4Native(enc, key.toByteArray())
        //        byte[] dec2 = RC4Utils.rc4(dec, key.getBytes());

        Log.e("jni enc", Arrays.toString(dec))
        Log.e("jni", String(dec))
    }
}
